[Bug] Example doesn't compile

**Describe the bug**

`go run worker/main.go` won't worker and neither `go build worker/main.go`

The main reason for this is that in go you can't pass a function of a struct to anything essentially.

this line https://github.com/temporalio/temporal-ecommerce/blob/main/worker/main.go#L23 will never compile.

```
ep@macbook-pro temporal-ecommerce % go run worker/main.go
# command-line-arguments
worker/main.go:23:21: undefined: app.CreateStripeCharge
worker/main.go:24:21: undefined: app.SendAbandonedCartEmail
```

This example is really helpful in general as it serves as a reference to how design your application but it would be great to have a working one as well. Thanks again!

--- a/worker/main.go
+++ b/worker/main.go
@@ -20,8 +20,11 @@ func main() {
 	// This worker hosts both Worker and Activity functions
 	w := worker.New(c, "CART_TASK_QUEUE", worker.Options{})
 
-	w.RegisterActivity(app.CreateStripeCharge)
-	w.RegisterActivity(app.SendAbandonedCartEmail)
+	var a *app.Activities
+
+	w.RegisterActivity(a.CreateStripeCharge)
+	w.RegisterActivity(a.SendAbandonedCartEmail)
+
 	w.RegisterWorkflow(app.CartWorkflow)
 	// Start listening to the Task Queue
 	err = w.Run(worker.InterruptCh())

